modules/platforms/cpp/core/src/impl/ignite_environment.cpp

```diff
@@ -118,6 +119,34 @@ namespace ignite
             }
         };
 
+        /*
+         * Stores pointer to internal Ignite node.
+         */
+        class IgniteNodeHolder
+        {
+        public:
+            IgniteNodeHolder(IgniteEnvironment* env) :
+                node(new IgniteImpl(SharedPointer<IgniteEnvironment>(env, IgniteNodeHolder::EmptyDeleter)))
+            {
+                // No-op.
+            }
+
+            Ignite* GetIgnite()
+            {
+                return &node;
+            }
+
+        private:
+            static void EmptyDeleter(IgniteEnvironment*)
```

Comment:
I like the decision, but let's move this function to common::utils and make it general, like follows:
```
template<typename T>
void EmptyDeleter(T*)
{
   // No-op.
}
```

Then we can get rid of `IgniteNodeHolder` altogether.

Reply:
Fixed.